models/classes/runner/QtiRunnerService.php

```diff
@@ -1795,7 +1793,7 @@ protected function getTransmissionId(QtiRunnerServiceContext $context, $itemId =
      * @param RunnerServiceContext $context
      * @throws \common_exception_InvalidArgumentType
      */
-    protected function assertQtiRunnerServiceContext(RunnerServiceContext $context)
+    public function assertQtiRunnerServiceContext(RunnerServiceContext $context)
```

Comment:
Shouln't the `OfflineQtiRunnerService` inherit from `QtiRunnerService`, since they are sharing a lot of logic?

Reply:
I was planning to handle them as separate services to be able to extend it by other extensions more easily.